wandb/wandb
# fix(launch): More generic error message + thread handling when Docker fails

Fixes [WB-12704](https://wandb.atlassian.net/browse/WB-12704) (It turns out a [previous PR](https://github.com/wandb/wandb/pull/5092) fixes this, but the error message was misleading)

Description
-----------
Two changes:
1. Our current Docker error message assumes it failed because the Docker daemon wouldn't connect. It could actually fail for multiple reasons, and Docker will output details about what the issue is. So, the message is more generic
2. When a Docker error occurred, we weren't gracefully terminating the thread the same way we do with other errors. This adds that.

Testing
-------
- In Docker Desktop -> Settings -> Resources, set Virtual disk limit to 8GB
- Start an agent on your machine
- Add a job to the queue the agent is polling based on this run: https://wandb.ai/tim-hays/debugging/runs/n0w5qt0h
- Notice the new error message coming from the launch agent, and that the agent goes back to polling after the run fails to start
![image](https://user-images.githubusercontent.com/39655500/224087274-50380b5b-d1da-4dab-ac7f-6a55095f3648.png)


[WB-12704]: https://wandb.atlassian.net/browse/WB-12704?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
